src/map/graphql.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  MapCollectionSearchResult,
+  MapLayerSearchResult,
+  QueryMapCollectionSearchArgs,
+  QueryMapLayerSearchArgs,
+  ResolverFn,
+} from '../generated/graphql'
+import { DEFAULT_LIMIT } from '../graphql/config'
+import getPageInfo from '../graphql/utils/getPageInfo'
+import paginate from '../utils/paginate'
+import { createMapCollectionsFuse, createMapLayersFuse } from './data'
+
+const mapCollectionsFuse = createMapCollectionsFuse([
+  'title',
+  'mapLayers.title',
+  'mapLayers.meta.description',
+  'mapLayers.themes.title',
+])
+const mapLayersFuse = createMapLayersFuse(['title', 'meta.description', 'themes.title'])
+
+export const mapCollectionSearch: ResolverFn<
+  MapCollectionSearchResult,
+  {},
+  any,
+  QueryMapCollectionSearchArgs
+> = (_, { q: query, input }): MapCollectionSearchResult => {
+  const page = input?.page ?? 1
+  const limit = input?.limit ?? DEFAULT_LIMIT
+  const results = mapCollectionsFuse.search(query)
```

Comment:
If I use an empty string for `q` it won't return any results, so how to get all the collections or maplayers?

Reply:
Change this so that now the query is optional, if it's optional it will return all results instead.